### PR TITLE
perf: add top level symbol

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: crates/rolldown/src/utils/renamer.rs [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 use rolldown_utils::rayon::{IntoParallelIterator, IntoParallelRefIterator, ParallelIterator};
-use rustc_hash::{FxHashMap, FxHashSet};
+use rustc_hash::FxHashMap;
 
 use crate::types::symbols::Symbols;
 
 #[derive(Debug)]
 pub struct Renamer<'name> {
-  used_canonical_names: FxHashSet<Cow<'name, Rstr>>,
+  pub used_canonical_names: FxHashMap<Cow<'name, Rstr>, u32>,
```
Comment (Boshen) [inline crates/rolldown/src/utils/renamer.rs:15]: no `pub` please!

Comment (Boshen) [reply crates/rolldown/src/utils/renamer.rs:15]: You should also comment what `u32` is

e.g. `/* count in a ... */ u32`

File: crates/rolldown/src/utils/renamer.rs [hunk 2]
```diff
@@ -66,40 +74,41 @@ impl<'name> Renamer<'name> {
...
           },
-          std::collections::hash_map::Entry::Occupied(_) => {
+          Entry::Occupied(_) => {
             // The symbol is already renamed
           }
         }
       });
 
       stack.push(Cow::Owned(used_canonical_names_for_this_scope));
+      // dbg!(&stack);
```
Comment (Boshen) [inline crates/rolldown/src/utils/renamer.rs:111]: extra comment here

File: crates/rolldown/src/utils/renamer.rs [hunk 3]
```diff
@@ -39,17 +40,24 @@ impl<'name> Renamer<'name> {
...
+      Entry::Vacant(vacant) => {
         let mut candidate_name = original_name.clone();
-        while self.used_canonical_names.contains(&candidate_name) {
-          count += 1;
-          candidate_name = Cow::Owned(format!("{original_name}${count}").into());
+        match self.used_canonical_names.entry(original_name.clone()) {
+          Entry::Occupied(mut occ) => {
+            let next_conflict_index = *occ.get() + 1;
+            *occ.get_mut() = next_conflict_index;
+            candidate_name = Cow::Owned(format!("{candidate_name}${next_conflict_index}",).into());
```
Comment (Boshen) [inline crates/rolldown/src/utils/renamer.rs:49]: extra `,` here.

**Comment by IWANABETHATGUY** [comment]
see https://github.com/rolldown/rolldown/pull/1461

**Pull request closed without merging**